src/v/cluster_link/replication/deps_impl.cc

```diff
@@ -111,6 +115,141 @@ remote_partition_source::fetch_next(ss::abort_source& as) {
       .batches = std::move(batches), .units = std::move(units)};
 }
 
+ss::future<kafka::offset> remote_partition_source::fetch_starting_offset() {
```

Comment:
nit: feel like this should have timeout or some such (if the remote is dead or this node has a network partition or some such).